src/edu/ucsb/nceas/metacat/dataone/MNodeService.java

```diff
@@ -449,6 +449,12 @@ public Identifier update(Session session, Identifier pid, InputStream object,
         NodeReference originMemberNode = this.getCapabilities().getIdentifier();
         sysmeta.setOriginMemberNode(originMemberNode);
 
+        // if no authoritative MN, set it to the same
+        if (sysmeta.getAuthoritativeMemberNode() == null || sysmeta.getAuthoritativeMemberNode().getValue().trim().equals("") ||
```

Comment:
I think I'd rather see just a null-check here instead so this conditional is just checking whether `authoritativeMemberNode` is set rather than set to something sensible. `authoritativeMemberNode` is defined in the schema with as a `d1:NonEmptyString` so, for one, "null" is a valid node identifier and second, schema validation (which would be done elsewhere if we even do it) will catch the empty-string case.

Reply:
ah :) I see you modeled this after d208e38cf89ea453f81c714766efd6fc5848444a. Carry on. I'll defer to @taojing2002 on that.